tests/integration_test_net.go

```diff
@@ -44,54 +47,112 @@ import (
 // integration test networks can also be used for automated integration and
 // regression tests for client code.
 type IntegrationTestNet struct {
-	done      <-chan struct{}
-	validator Account
+	done       <-chan struct{}
+	Validator  Account
+	clientPort int
+}
+
+func isPortFree(host string, port int) bool {
+	address := fmt.Sprintf("%s:%d", host, port)
+	listener, err := net.Listen("tcp", address)
+	if err != nil {
+		return false
+	}
+	listener.Close()
+	return true
+}
+
+func getFreePort() (int, error) {
+	var port int
+	for i := 0; i < 10; i++ {
```

Comment:
should `10` be a constant with a more representative name, or even a configurable parameter ?

Reply:
let me see what can we do, this technique has been used in norma already.